services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/RouteOptions.java

```diff
@@ -844,6 +845,62 @@ public static RouteOptions fromUrl(@NonNull URL url) {
     return fromJsonString(optionsJson.toString());
   }
 
+  /**
+   * Create a URL from RouteOptions instance.
+   *
+   * @param accessToken access token to make API request
+   * @return a URL object instance
+   */
+  @NonNull
+  public URL toUrl(@NonNull String accessToken) {
+    StringBuilder sb = new StringBuilder();
+    sb.append(baseUrl());
+    sb.append("/directions/v5");
+    sb.append(String.format("/%s", user()));
+    sb.append(String.format("/%s", profile()));
+    sb.append(String.format("/%s", coordinates()));
+    sb.append(String.format("?access_token=%s", accessToken));
+    sb.append(String.format("&alternatives=%s", alternatives()));
+    sb.append(String.format("&geometries=%s", geometries()));
+    sb.append(String.format("&overview=%s", overview()));
+    sb.append(String.format("&radiuses=%s", radiuses()));
+    sb.append(String.format("&steps=%s", steps()));
+    sb.append(String.format("&bearings=%s", bearings()));
+    sb.append(String.format("&layers=%s", layers()));
+    sb.append(String.format("&continue_straight=%s", continueStraight()));
+    sb.append(String.format("&annotations=%s", annotations()));
+    sb.append(String.format("&language=%s", language()));
+    sb.append(String.format("&roundabout_exits=%s", roundaboutExits()));
+    sb.append(String.format("&voice_instructions=%s", voiceInstructions()));
+    sb.append(String.format("&banner_instructions=%s", bannerInstructions()));
+    sb.append(String.format("&voice_units=%s", voiceUnits()));
+    sb.append(String.format("&exclude=%s", exclude()));
+    sb.append(String.format("&include=%s", include()));
+    sb.append(String.format("&approaches=%s", approaches()));
+    sb.append(String.format("&waypoints=%s", waypointIndices()));
+    sb.append(String.format("&waypoint_names=%s", waypointNames()));
+    sb.append(String.format("&waypoint_targets=%s", waypointTargets()));
+    sb.append(String.format("&enable_refresh=%s", enableRefresh()));
+    sb.append(String.format("&walking_speed=%s", walkingSpeed()));
+    sb.append(String.format("&walkway_bias=%s", walkwayBias()));
+    sb.append(String.format("&alley_bias=%s", alleyBias()));
+    sb.append(String.format("&snapping_include_closures=%s", snappingIncludeClosures()));
+    sb.append(String.format("&arrive_by=%s", arriveBy()));
+    sb.append(String.format("&depart_at=%s", departAt()));
+    sb.append(String.format("&max_height=%s", maxHeight()));
+    sb.append(String.format("&max_width=%s", maxWidth()));
+    sb.append(String.format("&metadata=%s", metadata()));
+
+    URL url;
+    try {
+      url = new URL(sb.toString());
+    } catch (MalformedURLException ex) {
+      throw new RuntimeException(ex);
+    }
+
+    return url;
```

Comment:
just curious if it would work
```suggestion
    try {
      retrun new URL(sb.toString());
    } catch (MalformedURLException ex) {
      throw new RuntimeException(ex);
    }
```